### ISSUE TITLE
Possible to select FP16, FP32 or FP64 to analyse training duration and precision? 

### ISSUE THREAD

**Issue by pusj** [jaungiers/LSTM-Neural-Network-for-Time-Series-Prediction#6]
Hello @jaungiers ,

I am really impressed by your work, a few weeks ago I downloaded your LSTM model and corrected it for Tensorflow 1.0 / Python 3.5 (range not equal to xrange and so fourth...)...and now you have done the same :-)

I was wondering if you have looked into using different data types...? Since GPU computing offers the possibility to train models quite fast and especially when using data types that require less storage memory (e.g. FP16) it would be a very nice addition to be able to experiment with different data types. This would not only be to determine the potential increase in calculation speed but also to analyse the effects on model accuracy (for this, the random numbers should be seeded so that the weight and bias matrices are identical for each run). Do you think data type selection something that could easily be added to this project?

Please let me know if you have any questions for me, I'd be happy to assist with further thoughts and explanations. I recently started using tensorflow after struggling with Matlab quite a bit. (https://se.mathworks.com/matlabcentral/answers/285851-suggestions-for-improvement-s-on-narnet-multistep-ahead-predictions-on-the-solar-dataset)

Best Regards
Staffan

**Comment by jaungiers** [comment]
Hey Staffan,

Thanks for your response. I haven't actually looked into using different data types yet but it's certainly an interesting suggestion! I shall do some research on using half-precision floating points in Python and then maybe try it out on some larger, deeper neural nets and get back to you. At the very least it should save me some computation time, so thank you :-) :+1:

Jakob

**Comment by pusj** [comment]
Hello again @jaungiers ,

Thank you very much for this quick and sympathetic answer. 

There are many benefits of having knowledge of data types, to be able utilize the full performance of the GPU; single precision (FP32) should be used. We are on a good path, I can confirm that the GPU is working with 80-90 % of the processing resource in training of the LSTM's which is good.

My thinking is that instead of starting diving into applying different settings (like the ones done in this thread [https://github.com/tensorflow/tensorflow/issues/1300](url)) my idea is just to test the FP16, FP32 and FP64 performance and see what we get. By this we can indirectly confirm that we are using the GPU and LSTM's in the correct way. 

I would be very happy if I could include this myself in the code, however, I am still at the point where I'm struggling with disabling the GPU (to only use the CPU, as described here: http://stackoverflow.com/questions/37660312/run-tensorflow-on-cpu) 

I don't know if this is any help for you but here is an overview of data types used in tensorflow: https://www.tensorflow.org/programmers_guide/dims_types   

My past experiences using GPU's in to speed up training in Matlab were not a huge success (https://se.mathworks.com/matlabcentral/answers/291744-time-series-prediction-using-neural-networks-narnet-narxnet-is-it-at-all-possible-to-train-a-ne ), even though it was very easy to chose between data types and feed the matricies to the GPU I never obtained any speedup. The only comment I got from Mathworks something similar to:  _"Yes, we know it's slow."_
